spark/src/main/java/org/apache/iceberg/spark/TypeToSparkType.java

```diff
@@ -104,12 +104,7 @@ public DataType primitive(Type.PrimitiveType primitive) {
         throw new UnsupportedOperationException(
             "Spark does not support time fields");
       case TIMESTAMP:
-        Types.TimestampType timestamp = (Types.TimestampType) primitive;
-        if (timestamp.shouldAdjustToUTC()) {
-          return TimestampType$.MODULE$;
-        }
-        throw new UnsupportedOperationException(
-            "Spark does not support timestamp without time zone fields");
```

Comment:
Could we do the flag check here as well to check whether or not we have enabled the "Handle without timezone" flag here as well? We may be using this not on the read path (like in the migrate/snapshot code) and it would be good to catch it here as well and make sure users know what is happening.

Reply:
I think this might involve a bigger refactor, including changing the method signature to accept the flag to `.primitive(..., handleTimestampWithoutTimezoneFlag)`. I'm not sure if that will break other stuff. I added in the logic as comments for now as to how to implement it after we can settle on the implementation

Reply:
I think that's fair, I can always fix this in the migrate code directly